woocommerce/woocommerce-android
# Issue/6957 upsell card reader banner in order list screen

Closes: #6957 


### Description

This PR adds upsell card reader banner as well as the dismiss alert dialog on the order list screen. 

### Note to reviewer
https://github.com/woocommerce/woocommerce-android/pull/6981 does some refactoring of the business logic of the upsell card reader banner. I would suggest referring to that PR to better understand how the code evolves and just to keep in mind that the business logic in this PR gets refactored in the later PR.

### Testing instructions

1. Navigate to the order list screen
2. Notice the banner
3. Play around by tapping dismiss (Remind me later, Don't show again) and `purchase card reader` options
4. Ensure the purchase card reader opens the appropriate link when in the US (M2 reader) and CA(WisePad 3 reader) stores.

### Images/gif


https://user-images.githubusercontent.com/1331230/179737056-d354b660-67c9-48f2-8c96-3d21ab90a7ef.mp4
